sandbox.go

```diff
@@ -80,6 +80,7 @@ type sandbox struct {
 	ingress            bool
 	ndotsSet           bool
 	sync.Mutex
+	sync.RWMutex
```

Comment:
You adding `sync.RWMutex` along with the existing `sync.Mutex`.
From which package will all the existing calls  `sb.Lock()/Unlock()` execute, `sync.mutex` or `sync.RWMutex` ?
If we were to accept this change, you should remove `sync.Mutex`.